x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credentials_form/gcp_credentials_form_agentless.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiSpacer } from '@elastic/eui';
+
+import {
+  GcpFormProps,
+  GCPSetupInfoContent,
+  GcpInputVarFields,
+  gcpField,
+  getInputVarsFields,
+} from './gcp_credential_form';
+import { getPosturePolicy } from '../utils';
+import { ReadDocumentation } from '../aws_credentials_form/aws_credentials_form';
+import { cspIntegrationDocsNavigation } from '../../../common/navigation/constants';
+
+export const GcpCredentialsFormAgentless = ({
+  input,
+  newPolicy,
+  updatePolicy,
+  disabled,
+}: GcpFormProps) => {
+  const accountType = input.streams?.[0]?.vars?.['gcp.account_type']?.value;
+  const isOrganization = accountType === 'organization-account';
+
+  /*
+    For Agentless only JSON credentials type is supported.
+    Also in case of organisation setup, project_id is not required in contrast to Agent-based.
+   */
+  const fields = getInputVarsFields(input, gcpField.fields).filter((field) => {
+    if (isOrganization) {
+      return ['gcp.organization_id', 'gcp.credentials.json'].includes(field.id);
+    } else {
+      return ['gcp.project_id', 'gcp.credentials.json'].includes(field.id);
+    }
+  });
```

Comment:
I think we could extract these GCP fields outside the filter loop to improve the readability:

```typescript
  const organizationFields = ['gcp.organization_id', 'gcp.credentials.json'];
  const singleAccountFields = ['gcp.project_id', 'gcp.credentials.json'];

	const gcpFields = isOrganization ? organizationFields : singleAccountFields;

	const fields = getInputVarsFields(input, gcpField.fields).filter(field => gcpFields.includes(field.id));
```

Reply:
Also, we should at some point convert those magic strings like `gcp.organization_id`, `gcp.credentials.json`, `gcp.account_type` and `gcp.project_id` into constants, we can address that in a separate PR

Reply:
That makes sense, will fix it!